solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m

```diff
@@ -0,0 +1,49 @@
+classdef Cartesian2CylindricalTransform < drakeFunction.DrakeFunction
+  % Given a cartesian coordinate frame, transform its coordinates to a
+  % cylindrical coordinates
+  properties
+    T_cylinder % This is a homogeneous transformation matrix, T_cylinder transforms a cylinder with [0;0;1] being its axis, and [1;0;0] as its theta=0 radius, to the cylinder in the cartesian frame
+  end
+  
+  methods
+    function obj = Cartesian2CylindricalTransform(T_cylinder)
+      input_frame = drakeFunction.frames.realCoordinateSpace(6);
+      output_frame = drakeFunction.frames.realCoordinateSpace(6);
+      obj = obj@drakeFunction.DrakeFunction(input_frame,output_frame);
+      if(~isHT(T_cylinder))
+        error('T_cylinder should be a homogenous transformation matrix');
+      end
+      obj.T_cylinder = T_cylinder;
+    end
+    
+    function [x_cylinder,J] = eval(obj,x_cartesian)
+      % x_cylinder = [radius;theta;height;roll;pitch;yaw]
+      % we attach a cartesian frame [x_tangent;y_normal;z_tangent] in R^3x3 at the tip of the
+      % radius, where x_tangent is aligned with cross(height,radius),
+      % y_norm is along the radius direction, and z_tangent =
+      % cross(x_tangent,y_normal). roll;pitch;yaw are the Euler angles that
```

Comment:
investigated this, and yes it matches as long as x_tangent is indeed cross(radius, height)